middleware/etcd/msg/type_test.go

```diff
@@ -0,0 +1,31 @@
+package msg
+
+import (
+	"testing"
+
+	"github.com/miekg/dns"
+)
+
+func TestType(t *testing.T) {
+	tests := []struct {
+		serv         Service
+		expectedType uint16
+	}{
+		{Service{Host: "example.org"}, dns.TypeANY},
+		{Service{Host: "127.0.0.1"}, dns.TypeA},
+		{Service{Host: "2000::3"}, dns.TypeAAAA},
+		{Service{Host: "2000..3"}, dns.TypeANY},
+		{Service{Host: "127.0.0.257"}, dns.TypeANY},
+		{Service{Host: "127.0.0.252", Mail: true}, dns.TypeA},
+		{Service{Host: "127.0.0.252", Mail: true, Text: "a"}, dns.TypeA},
```

Comment:
@miekg, does this mean A record could also be a MX record and a TXT record?
Slightly difficult to comprehend. I seriously hoped Service contained a field specifically to signify the record type.